nixos/modules/services/hardware/amdgpu.nix

```diff
@@ -0,0 +1,45 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.amdgpu;
+in {
+  options.hardware.amdgpu = {
+    legacySupport.enable = lib.mkEnableOption ''
+      support for old AMD graphics cards, this uses `amdgpu`
+      instead of `radeonsi` driver for Southern Islands (Radeon HD 7000)
+      series and Sea Islands (Radeon HD 8000) series cards.
+      Note: this removes support for analog video output graphics cards,
+      which is only available in `radeonsi` driver
+    '';
+    initrd.enable = lib.mkEnableOption ''
+      loading `amdgpu` kernelModule in stage 1.
+      Can fix lower resolution in boot screen during initramfs phase
+    '';
+    rocm.enable = lib.mkEnableOption ''ROCm/OpenCL support'';
+    # cfg.amdvlk option is defined in ./amdvlk.nix module
+  };
+
+  config = {
+    boot.kernelParams = lib.optionals cfg.legacySupport.enable [
+      "amdgpu.si_support=1"
+      "amdgpu.cik_support=1"
+      "radeon.si_support=0"
+      "radeon.cik_support=0"
+    ];
+
+    boot.initrd.kernelModules = lib.optionals cfg.initrd.enable [ "amdgpu" ];
```

Comment:
This should probably be optional.

Installing the amdgpu module in the initrd on my laptop causes a noticeable modeset, the TTY to wrap in a weird way and the brightness to jump to 100% which is why I've opted to explicitly not do this.

I know this is an edge-case but removing stuff from lists in a module is really annoying.

Reply:
This *is* marked as optional and not enabled by default, no?

Reply:
Apparently, I'm blind.